wren-ai-service/src/pipelines/sql_regeneration/generation.py

```diff
@@ -0,0 +1,215 @@
+import logging
+import sys
+from pathlib import Path
+from typing import Any, Dict, List, Optional
+
+import orjson
+from hamilton import base
+from hamilton.experimental.h_async import AsyncDriver
+from haystack import component
+from haystack.components.builders.prompt_builder import PromptBuilder
+from langfuse.decorators import observe
+
+from src.core.engine import Engine
+from src.core.pipeline import BasicPipeline
+from src.core.provider import LLMProvider
+from src.pipelines.common import GenerationPostProcessor
+from src.pipelines.sql_regeneration.components.prompts import (
+    sql_regeneration_system_prompt,
+)
+from src.utils import async_timer, timer
+from src.web.v1.services.sql_regeneration import (
+    SQLExplanationWithUserCorrections,
+)
+
+logger = logging.getLogger("wren-ai-service")
+
+
+sql_regeneration_user_prompt_template = """
+inputs: {{ results }}
+
+Let's think step by step.
+"""
+
+
+@component
+class SQLRegenerationRreprocesser:
+    @component.output_types(
+        results=Dict[str, Any],
+    )
+    def run(
+        self,
+        description: str,
+        steps: List[SQLExplanationWithUserCorrections],
+    ) -> Dict[str, Any]:
+        return {
+            "results": {
+                "description": description,
+                "steps": steps,
+            }
+        }
+
+
+@component
+class DescriptionRegenerationPostProcessor:
+    @component.output_types(
+        results=Optional[Dict[str, Any]],
+    )
+    def run(
+        self,
+        replies: List[str],
+        steps: List[str],
+    ) -> Dict[str, Any]:
+        try:
+            return {
+                "results": {
+                    "description": orjson.loads(replies[0]).get("description", ""),
+                    "steps": steps,
+                }
+            }
+        except Exception as e:
+            logger.exception(f"Error in DescriptionRegenerationPostProcessor: {e}")
+            return {"results": None}
```

Comment:
Curiously, it isn't used anywhere. Why do we keep it?

Reply:
removed